File: frontend/src/app/cliente/shared/service/cliente.service.ts
import { ContatoCliente } from './../entidade/contatoCliente';
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Cliente } from '../entidade/cliente';
import { Automovel } from '../entidade/automovel';
import { Imovel } from '../entidade/imovel';

@Injectable({
  providedIn: 'root'
})
export class ClienteService {

  constructor(private http: HttpClient) { }

  public cadastrarCliente(clienteDTO: Cliente) {
    return this.http.post<any>("http://localhost:8080/cliente/cadastrar", clienteDTO);
  }

  public cadastrarSeguroAutomovel(cpfCliente: string, seguroVeiculoDTO: Automovel) {
    return this.http.post<any>("http://localhost:8080/seguro/veiculo/"+cpfCliente, seguroVeiculoDTO);
  }

  public cadastrarSeguroImovel(cpfCliente: string, seguroImovelDTO: Imovel) {
    return this.http.post<any>("http://localhost:8080/seguro/imovel/"+cpfCliente, seguroImovelDTO);
  }

  public listarClientes() {
    return this.http.get<any>("http://localhost:8080/cliente/buscar/lista");
  }

  public excluirCliente(cliente : Cliente) {
    return this.http.delete<any>("http://localhost:8080/cliente/excluir/"+cliente.cpf);
  }

  public atualizarCliente(cliente : Cliente) {
    return this.http.put<any>("http://localhost:8080/cliente/atualizar/", cliente);
  }

  public atualizarContato(contatoCliente : ContatoCliente) {
    return this.http.put<any>("http://localhost:8080/cliente/atualizar/contato", contatoCliente);
  }
}

File: benfica-api/backend/src/main/java/com/benfica/service/ClienteService.java
package com.benfica.service;

import java.util.List;

import com.benfica.dto.AtualizarClienteDTO;
import com.benfica.dto.AtualizarContatoDTO;
import com.benfica.dto.ClienteDTO;
import com.benfica.dto.ClienteRetornoCompletoDTO;
import com.benfica.model.Cliente;

public interface ClienteService {

	Cliente buscar(Long id);
	Cliente cadastrar(ClienteDTO clienteDTO);
	String atualizar(AtualizarClienteDTO atualizarClienteDTO);
	String excluir(String cpf);
	Cliente buscarPorCpf(String cpf);
	List<ClienteRetornoCompletoDTO> buscarLista();
	String atualizarContato(AtualizarContatoDTO atualizarContatoDTO);
}
